pkg/controller/common.go

```diff
@@ -528,12 +530,30 @@ func getQualifiedName(namespace, name string) string {
 	return fmt.Sprintf("%s/%s", namespace, name)
 }
 
-func formatUpdateStateError(objectType, fromState string, possibleStates ...string) string {
-	length := len(possibleStates)
-	if length > 1 {
-		possibleStates[length-1] = fmt.Sprintf("or %s", possibleStates[length-1])
+func formatUpdateStateError(objectType, fromState, toState string, expectedStates ...string) string {
+	return fmt.Sprintf("%s's state '%s' cannot be updated to %s. %s can only be updated to %s", objectType, fromState, toState, fromState, strings.Join(expectedStates, ", "))
+}
+
+func getOperationRequeueError(desired string, obj client.Object) error {
+	return status.Errorf(codes.Aborted, "requeueing %s operation because another opeation is already pending on %v (%s)", desired, reflect.TypeOf(obj), obj.GetName())
+}
+
+func reconcileReturnOnError(err error) (reconcile.Result, error) {
+	if status.Code(err) == codes.FailedPrecondition {
+		klog.Errorf("%v", err)
```

Comment:
Printing out the error w/o understanding the context makes debugging failures challenging. Perhaps pass in an operation name and object type and name so that we can print an error message like "failed to _{attach|detach|create|delete|update}_ _{AzVolume|AzVolumeAttachment}_ _name_: _error_".